servicetalk-transport-api/src/main/java/io/servicetalk/transport/api/IoExecutor.java

```diff
@@ -38,4 +38,12 @@
      * @return {@code true} if supported
      */
     boolean isFileDescriptorSocketAddressSupported();
+
+    /**
+     * Returns true if the provided thread is an IOExecutor thread.
+     *
+     * @param thread the thread of interest
+     * @return {@code true} if an IOExecutor thread otherwise {@code false}.
+     */
+    boolean isIoExecutorThread(Thread thread);
```

Comment:
We shouldn't need this method and instead we should move this as an implementation detail of the `netty` packages. I think this can be accomplished via the `HttpServerBuilder` (which lives in `http.api`, and will apply the offloading filter in https://github.com/apple/servicetalk/pull/1710) accepting a `Predicate<Thread>` which is translated into the `BooleanSupplier` via:

```java
Predicate<Thread> predicate = thread -> thread instanceof AsyncContextHolderNettyThread;
BooleanSupplier shouldOffload = () -> predicate.test(Thread.currentThread());
```

Also the granularity of this method is off as the `Thread` may or may not be known to this `IoExecutor`, so a single instance of `IoExecutor` won't be authoritative on all `Threads.

Reply:
Yes, something similar (or exactly) is the transition plan for this API. The method on IoExecutor is temporary.

Reply:
I have marked it as deprecated in the draft PR.

Reply:
If this change is easier to accomplish in combination with changes from another PR than can we combine them or sequence the changes differently (e.g. this change built on top of the other)? I don't think it makes sense to have this temporary API merged in, and it would be easier to review/evaluate the end state with out temporary states.

Reply:
The next draft PR is base upon #1710 and now uses the `IoThreadFactory` static method via `NettyHttpServerBuilder`